src/axom/inlet/examples/document_generation_example.cpp

```diff
@@ -0,0 +1,173 @@
+// usage : ./document_generation_example_example --enableDocs --deck lua_file --verbose --example [example #]
+// example 1: field1 = true; field2 = 5632; NewTable = { str = 'hello'; integer = 32 }
+// example 2: foo = false; bar = true; Table1 = { float1 = 3.14; Table11 = { Table111 = { x = 4 } } }
+// example 3: Table1 = { float1 = 5.6 }; Table2 = { int1 = 95 }; Table3 = { bool1 = true }
+
+#include "axom/inlet/DocWriter.hpp"
+#include "axom/inlet/Inlet.hpp"
+#include "axom/inlet/LuaReader.hpp"
+
+#include "CLI11/CLI11.hpp"
+#include <iostream>
+
+using axom::inlet::Inlet;
+using axom::inlet::LuaReader;
+using axom::sidre::DataStore;
+using axom::inlet::DocWriter;
+
+void createInletEx1(std::shared_ptr<Inlet> inlet)

```

Comment:
This is a good test/example but should probably be broken up into a new set of tests and then leave behind a clean and concise example for users to emulate.